pkg/reconciler/mtbroker/controller.go

```diff
@@ -109,5 +106,29 @@ func NewController(
 		},
 	))
 
+	// When the endpoints in our multi-tenant filter/ingress change, do a global resync.
+	// During installation, we might reconcile Brokers before our shared filter/ingress is
+	// ready, so when these endpoints change perform a global resync.
```

Comment:
Why do we need a global resync instead of just enqueuing brokers? The way I read this "installation..." part is that even after reaching a steady state, if endpoints change, the brokers wouldn't be enqueued? Assuming that all the Brokers are using MT based one, this is fine since they would all need to get enqueued anyways, but just curious why simply enqueing affected brokers is not enough?
Seems like we should enqueue all Brokers that have the MTChannelBasedBroker annotation only instead of all the Brokers?

Reply:
All the global resync does is enqueue brokers.  This is a *filtered* global resync, so it will only enqueue the MT brokers.

re: affected brokers

All of the MT brokers are affected by changes to the MT endpoint.  1->5 may not affect readiness, but 1->0 does (e.g. if the single pod crashes or is evicted) as would 0->1 (what I see)